### PR TITLE
Support HVDC AC emulation simulation in DC Load Flow

### PR DESCRIPTION
**Pull request by vidaldid-rte**
Alternative proposal for https://github.com/powsybl/powsybl-open-loadflow/pull/1009

### CODE REVIEW

File: src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java [hunk 1]
```diff
@@ -207,32 +207,6 @@ void testHvdcAcEmulationNonSupported() {
         assertReactivePowerEquals(-10.0, cs3.getTerminal());
     }
 
-    @Test
-    void testHvdcAcEmulationNonSupported2() {
```
Comment (annetill) [inline src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:211]: You don't want to keep it with a non zero P0 maybe?

Comment (vidaldid-rte) [reply src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:211]: I do not understand @annetill . non zero P0 is covered in testDcLoadFlowWithHvdcAcEmulation2 as well.
DId I remove a test that is not performed in the new one  ?

Comment (annetill) [reply src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:211]: Yes sorry, I see it now.

**Review by annetill — APPROVED** [review]

File: src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java [hunk 2]
```diff
@@ -529,4 +503,110 @@ void testAcEmuAndPMax() {
...
+
+        assertActivePowerEquals(16.482, cs2.getTerminal()); // 10MW + 180 MW/deg * 0.036008deg
+        assertAngleEquals(0.0, cs2.getTerminal().getBusView().getBus());
+        assertActivePowerEquals(-16.482, cs3.getTerminal());
+        assertAngleEquals(-0.036008, cs3.getTerminal().getBusView().getBus());
+
+        result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isFullyConverged());
+
+        assertActivePowerEquals(16.482, cs2.getTerminal()); // 0MW + 180 MW/deg * 0.045105deg
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:559]:
```suggestion
        assertActivePowerEquals(16.482, cs2.getTerminal()); // 10MW + 180 MW/deg * 0.036008deg
```

File: src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java [hunk 3]
```diff
@@ -529,4 +503,110 @@ void testAcEmuAndPMax() {
...
+        network.getHvdcLine("hvdc32").newExtension(HvdcAngleDroopActivePowerControlAdder.class)
+                .withDroop(180)
+                .withP0(0.f)
+                .withEnabled(true)
+                .add();
+
+        result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isFullyConverged());
+
+        assertActivePowerEquals(15.578, cs2.getTerminal()); // 0MW + 180 MW/deg * 0.045105deg
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:606]:
```suggestion
        assertActivePowerEquals(15.578, cs2.getTerminal()); // 10MW + 180 MW/deg * 0.030988deg
```

File: src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java [hunk 4]
```diff
@@ -529,4 +503,110 @@ void testAcEmuAndPMax() {
...
+        assertAngleEquals(0.0, cs2.getTerminal().getBusView().getBus());
+        assertActivePowerEquals(-15.578, cs3.getTerminal());
+        assertAngleEquals(-0.030988, cs3.getTerminal().getBusView().getBus());
+        assertActivePowerEquals(5.578, network.getVscConverterStation("cs2Bis").getTerminal()); // 0MW + 180 MW/deg * 0.045105deg
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:610]:
```suggestion
        assertActivePowerEquals(5.578, network.getVscConverterStation("cs2Bis").getTerminal()); // 0MW + 180 MW/deg * 0.030988deg
```

**Review by jeandemanged — CHANGES_REQUESTED** [review]
either remove or fix the test comments

File: src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java (hunk 4 above)
Comment (vidaldid-rte) [reply src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java:610]: Fixed. Thanks for catching it.

**Comment by jeandemanged** [comment]
@vidaldid-rte as mentionned in https://github.com/powsybl/powsybl-open-loadflow/pull/1013#pullrequestreview-1978979318 feel free to remove the comments if you feel these are unmaintainable.

**Review by jeandemanged — APPROVED** [review]

**Pull request merged**